enterprise/cypher/cypher/src/main/scala/org/neo4j/cypher/internal/compatibility/v3_3/runtime/interpreted/expressions/GetDegreePrimitive.scala

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright (c) 2002-2017 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.cypher.internal.compatibility.v3_3.runtime.interpreted.expressions
+
+import org.neo4j.cypher.internal.compatibility.v3_3.runtime.ExecutionContext
+import org.neo4j.cypher.internal.compatibility.v3_3.runtime.commands.expressions.Expression
+import org.neo4j.cypher.internal.compatibility.v3_3.runtime.pipes.QueryState
+import org.neo4j.cypher.internal.frontend.v3_3.SemanticDirection
+
+case class GetDegreePrimitive(offset: Int, typ: Option[String], direction: SemanticDirection) extends Expression {
+
+  override def apply(ctx: ExecutionContext)(implicit state: QueryState): Any = typ match {
+    case None => state.query.nodeGetDegree(ctx.getLongAt(offset), direction)
+    case Some(t) => state.query.getOptRelTypeId(t) match {
+      case None => 0
```

Comment:
instead of this second match, maybe use 
```
map(relTypeId => state.query.nodeGetDegree(ctx.getLongAt(offset), direction, relTypeId))
.getOrElse(0)
```

I don't think there is any difference in the produced bytecode, and it feels more readable IMHO.